todolist/file_store.go

```diff
@@ -36,45 +47,68 @@ func (f *FileStore) Initialize() {
 
 func (f *FileStore) Load() ([]*Todo, error) {
 	if f.FileLocation == "" {
-		f.FileLocation = getLocation()
+		var err error
+		f.FileLocation, err = getLocation()
+		if err != nil {
+			fmt.Println("Error whlie finding todo file", err)
+		}
 	}
 
 	data, err := ioutil.ReadFile(f.FileLocation)
 	if err != nil {
 		fmt.Println("No todo file found!")
 		fmt.Println("Initialize a new todo repo by running 'todo init'")
 		os.Exit(0)
-		return nil, err
 	}
 
 	var todos []*Todo
 	jerr := json.Unmarshal(data, &todos)
 	if jerr != nil {
 		fmt.Println("Error reading json data", jerr)
 		os.Exit(1)
-		return nil, jerr
 	}
 	f.Loaded = true
 
 	return todos, nil
 }
 
 func (f *FileStore) Save(todos []*Todo) {
+	os.MkdirAll(filepath.Dir(f.FileLocation), 0700)
+
 	data, _ := json.Marshal(todos)
 	if err := ioutil.WriteFile(f.FileLocation, []byte(data), 0644); err != nil {
 		fmt.Println("Error writing json file", err)
 	}
 }
 
-func getLocation() string {
-	localrepo := ".todos.json"
-	usr, _ := user.Current()
-	homerepo := fmt.Sprintf("%s/.todos.json", usr.HomeDir)
-	_, ferr := os.Stat(localrepo)
+func getLocation() (location string, err error) {
```

Comment:
Yes, it would be implicitly returned.

The [Bible](https://golang.org/doc/effective_go.html#named-results) suggests using this to simplify statements, but I think you are correct in this case - it would be clearer if the values were explicit